packages/@aws-cdk/aws-kms/lib/key.ts

```diff
@@ -193,6 +193,28 @@ abstract class KeyBase extends Resource implements IKey {
     return this.grant(grantee, ...[...perms.DECRYPT_ACTIONS, ...perms.ENCRYPT_ACTIONS]);
   }
 
+  /**
+   * Grant permissions to generating MACs to the given principal
+   */
+  public grantGenerateMac(grantee: iam.IGrantable): iam.Grant {
+    if (!(this.node.defaultChild as CfnKey).keySpec?.startsWith('HMAC')) {
```

Comment:
Is this really the best way to access the keySpec here? Seems like there should be an easier way to access what are essentially this instances own properties.

Reply:
You're right, this is not the correct way. And additionally this will not work for imported keys when I extend the Key interface, because the defaultChild has no `keySpec`.

I think the guard statement is not applicable here. For imported keys (e.g. `fromKeyArn`) we do not know the `keySpec` when instantiated. So to get this to work, I would have to completely redesign the `KeyBase` class, which is not the best choice here imho.

I will remove the guard statements. Let me know what you think.

Reply:
Yeah I suppose that is the best route here.